src/test/db.rs

```diff
@@ -362,3 +370,24 @@ async fn index_option_defaults_test(defaults: Option<IndexOptionDefaults>, name:
     };
     assert_eq!(event_defaults, defaults);
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn watch_test_db() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = EventClient::new().await;
+    if client.server_version_lt(4, 0) || client.is_standalone() {
+        return;
+    }
+    let db = client.database(function_name!());
+    let pipeline = vec![doc! { "$match": { "x": 3 } }];
+    let options = ChangeStreamOptions::builder()
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .read_concern(ReadConcern::from(ReadConcernLevel::Available))
```

Comment:
this test is specifying a read concern other than `Majority` on versions that include 4.0, so this test _should_ be failing. It appears to not be, though, meaning the read concern is likely not getting serialized properly. This needs investigation.